### ISSUE TITLE
llm_model valid_len for long audio

### ISSUE THREAD

**Issue by caofei88** [Wasser1462/FunASR-nano-onnx#2]
<img width="1930" height="1158" alt="Image" src="https://github.com/user-attachments/assets/60f1f3e8-d2ec-4d30-8af7-c41752bea5a9" />
 
if valid_len >= max_total_len:
                break
如何解决大音频转录
提高llm_onnx.py  119行代码性能 logits = self.model.run([self.output_name], ort_inputs)[0]  很耗时

**Comment by caofei88** [comment]
> 感谢反馈！这个脚本只是用于快速验证导出的 ONNX 输出的 demo，目前是全量重算的方式，没有针对长音频做性能优化。我实际推理llm已经改用 prefill/decode + KV cache，并用 C++ 在 CPU 上做推理，RTF 比较小。最近工作比较忙，等我把代码整理好会提交 PR 到 sherpa-onnx，后面可以关注一下。

感谢回复，期待哈！按照我的理解这里离线转录大文件音频需要对音频进行滑窗分段处理 + 拼接结果 ；**这个模型针对国外语种Fun-ASR-MLT-Nano-2512** 也能使用你验证脚本模型转onnx 格式吧

**Comment by xielinyun123** [comment]
大佬，你说的：
我实际推理llm已经改用 prefill/decode + KV cache，并用 C++ 在 CPU 上做推理，RTF 比较小
如果搞好了，可以更新下readme吗，一直在等你的更新
